### PR TITLE
zkApp composability, pt 2: multiple proofs

### PR DESCRIPTION
**Pull request by mitschabaude**
Small PR which fixes the case of multiple independent proofs in one transaction.
This isn't composability, but a case that came up on initial testing that turned out to be broken. The PR fixes it and adds an example which combines the proofs of two smart contracts together in one transaction block

The PR also has a couple of other minor changes, to collect unrelated stuff which would deter from reviewing the bigger chunks (see next PR)

### CODE REVIEW

File: src/examples/zkapps/simple_and_counter_zkapp.ts [hunk 1]
```diff
@@ -0,0 +1,249 @@
+/**
```
Comment (mitschabaude) [inline src/examples/zkapps/simple_and_counter_zkapp.ts:1]: the new example!

File: src/lib/party.ts [hunk 2]
```diff
@@ -657,8 +657,14 @@ class Party {
   }
 
   hash() {
-    let fields = Types.Party.toFields(toPartyUnsafe(this));
-    return Ledger.hashPartyFromFields(fields);
+    // these two ways of hashing are (and have to be) consistent / produce the same hash
```
Comment (mitschabaude) [inline src/lib/party.ts:660]: this is what fixes the new example. the previous version was always returning variables, even outside the circuit

**Review by MartinMinkov — APPROVED** [review]
LGTM!

File: src/examples/zkapps/simple_and_counter_zkapp.ts [hunk 3]
```diff
@@ -0,0 +1,249 @@
...
+
+    // compute the new counter and hash from pending actions
+    // remark: it's not feasible to pass in the pending actions as method arguments, because they have dynamic size
+    let { state: newCounter, actionsHash: newActionsHash } =
+      this.reducer.reduce(
+        offchainStorage.pendingActions,
+        // state type
+        Field,
+        // function that says how to apply an action
+        (state: Field, _action: Field) => {
```
[the remark this answers is not in the document]
Comment (MartinMinkov) [reply src/examples/zkapps/simple_and_counter_zkapp.ts:77]: Nvm, just realized that `actionsHash` is being passed in. I guess the use would be extra information we want to pass into the reducer function to do something specific?

**Pull request merged**